streams.h

```diff
@@ -218,6 +218,73 @@ struct sac_2d_all_pos : stream {
     std::size_t _flip_bit_position;
 };
 
+template<typename value_type>
```

Comment:
I just realized you shouldn't mix value_type with this: https://github.com/crocs-muni/eacirc-streams/blob/25de00081418995e63fd28a31a9a9c5550040844/stream.h#L9 definition.

Also, why are these function out of hw_counter structure? Why not as a private functions within hw_counter. Do you plan to use those somewhere else?